python/cudf/cudf/core/series.py

```diff
@@ -2621,15 +2643,57 @@ def get_dt_field(self, field):
         )
 
 
-def _align_indices(lhs, rhs, join="outer"):
+def _align_indices(series_list, how="outer", allow_non_unique=False):
     """
-    Internal util to align the indices of two Series. Returns a tuple of the
-    aligned series, or the original arguments if the indices are the same, or
-    if rhs isn't a Series.
+    Internal util to align the indices of a list of Series objects
+
+    series_list : list of Series objects
+    how : {"outer", "inner"}
+        If "outer", the values of the resulting index are the
+        unique values of the index obtained by concatenating
+        the indices of all the series.
+        If "inner", the values of the resulting index are
+        the values common to the indices of all series.
+    allow_non_unique : bool
+        Whether or not to allow non-unique valued indices in the input
+        series.
     """
-    if isinstance(rhs, Series) and not lhs.index.equals(rhs.index):
-        lhs, rhs = lhs.to_frame(0), rhs.to_frame(1)
-        lhs_rhs = lhs.join(rhs, how=join, sort=True)
-        lhs = lhs_rhs.iloc[:, 0]
-        rhs = lhs_rhs.iloc[:, 1]
-    return lhs, rhs
+    if len(series_list) <= 1:
+        return series_list
+
+    # check if all indices aer the same
+    head = series_list[0].index
+
+    all_index_equal = True
+    for sr in series_list[1:]:
+        if not sr.index.equals(head):
+            all_index_equal = False
+            break
+
+    if all_index_equal:
+        return series_list
+
+    if how == "outer":
+        combined_index = cudf.core.reshape.concat(
+            [sr.index for sr in series_list]
+        ).unique()
+    else:
+        combined_index = series_list[0].index
+        for sr in series_list[1:]:
+            combined_index = (
+                cudf.DataFrame(index=sr.index).join(
+                    cudf.DataFrame(index=combined_index),
+                    sort=True,
+                    how="inner",
+                )
+            ).index
+
+    # align all Series to the combined index
+    result = []
+    for sr in series_list:
+        result.append(
+            sr._align_to_index(
+                combined_index, how=how, allow_non_unique=allow_non_unique
+            )
+        )
```

Comment:
Use list comprehension instead of creating empty list and appending here

Reply:
Done